internal/provider/resource_hvn_route.go

```diff
@@ -1,14 +1,293 @@
+	}
+
+	log.Printf("[INFO] Created HVN route for HVN (%s) with destination CIDR %s", hvnRoute.ID, hvnRoute.Hvn.ID)
+
+	hvnRoute, err = clients.WaitForHVNRouteToBeActive(ctx, client, destination, hvnLink.ID, loc, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := setHVNRouteResourceData(d, hvnRoute, loc); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceHvnRouteRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	hvn := d.Get("hvn").(string)
+	var hvnLink *sharedmodels.HashicorpCloudLocationLink
+
+	hvnLink, err := parseLinkURL(hvn, HvnResourceType)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	destination := d.Get("destination_cidr").(string)
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	route, err := clients.ListHVNRoutes(ctx, client, hvnLink.ID, destination, "", "", loc)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			log.Printf("[WARN] HVN route for HVN (%s) with destination CIDR %s not found, removing from state", hvnLink.ID, destination)
+			d.SetId("")
+			return nil
+		}
+
+		return diag.Errorf("unable to retrieve HVN route for HVN (%s) with destination CIDR %s: %v", hvnLink.ID, destination, err)
+	}
+
+	if len(route) != 1 {
+		return diag.Errorf("Unexpected number of HVN routes returned when waiting for route with destination CIDR of %s for HVN (%s) to be Active: %d", destination, hvnLink.ID, len(route))
+	}
+
+	// The HVN route failed to provision properly so we want to let the user know and
+	// remove it from state.
+	if route[0].State == networkmodels.HashicorpCloudNetwork20200907HVNRouteStateFAILED {
+		log.Printf("[WARN] HVN route for HVN (%s) with destination CIDR %s failed to provision, removing from state", hvnLink.ID, destination)
+		d.SetId("")
+		return nil
+	}
+
+	// HVN route found, update resource data.
+	if err := setHVNRouteResourceData(d, route[0], loc); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceHvnRouteDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	link, err := buildLinkFromURL(d.Id(), HVNRouteResourceType, client.Config.OrganizationID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	routeID := link.ID
+	loc := link.Location
+
+	hvn := d.Get("hvn").(string)
+	hvnLink, err := buildLinkFromURL(hvn, HvnResourceType, loc.OrganizationID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] Deleting HVN route (%s)", routeID)
+	resp, err := clients.DeleteHVNRouteByID(ctx, client, hvnLink.ID, routeID, loc)
+
+	if err := clients.WaitForOperation(ctx, client, "delete HVN route", loc, resp.Operation.ID); err != nil {
+		if strings.Contains(err.Error(), "execution already started") {
```

Comment:
where do we return this error?

Reply:
Looks like this is a pattern from the peering resource, which had these comments:
```
		// If the HVN has already been deleted
		// TODO: Peerings can be deleted automatically by the network monitor workflow ]
		// and when deleting the HVN is deleted can cause an already started error.
		// It would be nicer if we could return a better error message or tie the operations together.
```

I'm not sure if the same thing applies to routes. Ideally we should remove the special case, or else come up with a new error message.

Reply:
currently backend will just return 503 for the second, third, etc delete requests of the same hvn route, I'll fix it so it would just return the same operation if delete is already in progress, either way this special case won't be necessary here